contracts/pair_anchor/src/contract.rs

```diff
@@ -0,0 +1,860 @@
+    DepsMut, Env, MessageInfo, Reply, Response, StdError, StdResult, Uint128,
+    
+    WasmMsg,
+};
+
+use crate::response::MsgInstantiateContractResponse;
+use astroport::asset::{addr_validate_to_lower, Asset, AssetInfo, PairInfo};
+use astroport::factory::PairType;
+
+use astroport::pair_anchor::{
+    AnchorExecuteMsg, ConfigResponse, DEFAULT_SLIPPAGE, MAX_ALLOWED_SLIPPAGE, AnchorQueryMsg, StateResponse, AnchorPoolParams,
+};
+use astroport::pair_anchor::{
+    CumulativePricesResponse, Cw20HookMsg, ExecuteMsg, MigrateMsg, PoolResponse,
+    QueryMsg, ReverseSimulationResponse, SimulationResponse,
+};
+use astroport::pair::{
+    InstantiateMsg
+};
+
+
+use cw2::set_contract_version;
+use cw20::{Cw20ExecuteMsg, Cw20ReceiveMsg};
+use protobuf::Message;
+use std::str::FromStr;
+use std::vec;
+
+/// Contract name that is used for migration.
+const CONTRACT_NAME: &str = "astroport-pair-anchor";
+/// Contract version that is used for migration.
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+// /// A `reply` call code ID used for sub-messages.
+// const INSTANTIATE_TOKEN_REPLY_ID: u64 = 1;
+
+// contract
+// terra1sepfj7s0aeg5967uxnfk4thzlerrsktkpelm5s
+// execute_msg
+// {
+//   "deposit_stable": {}
+// }
+// coins
+// [{"denom":"uusd","amount":"1000000"}]
+
+// contract
+// terra1hzh9vpxhsk8253se0vv5jj6etdvxu3nv8z07zu
+// execute_msg
+// {
+//   "send": {
+//     "amount": "822048",
+//     "contract": "terra1sepfj7s0aeg5967uxnfk4thzlerrsktkpelm5s",
+//     "msg": "eyJyZWRlZW1fc3RhYmxlIjp7fX0="
+//   }
+// }
+// {"redeem_stable":{}}
+// coins
+// []
+
+/// ## Description
+/// Creates a new contract with the specified parameters in the [`InstantiateMsg`].
+/// Returns the [`Response`] with the specified attributes if the operation was successful, or a [`ContractError`] if
+/// the contract was not created.
+/// ## Params
+/// * **deps** is an object of type [`DepsMut`].
+///
+/// * **env** is an object of type [`Env`].
+///
+/// * **_info** is an object of type [`MessageInfo`].
+/// * **msg** is a message of type [`InstantiateMsg`] which contains the basic settings for creating a contract.
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    msg.asset_infos[0].check(deps.api)?;
+    msg.asset_infos[1].check(deps.api)?;
+
+    if msg.asset_infos[0] == msg.asset_infos[1] {
+        return Err(ContractError::DoublingAssets {});
+    }
+
+    let params: AnchorPoolParams = from_binary(&msg.init_params.unwrap())?;
+
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    let config = Config {
+        pair_info: PairInfo {
+            contract_addr: env.contract.address.clone(),
+            liquidity_token: Addr::unchecked(""),
+            asset_infos: msg.asset_infos.clone(),
+            pair_type: PairType::Xyk {},
```

Comment:
Here I used Xyk pool, even though it would make sense to define a custom pool type for that similiar to the stable swap.

Reply:
Let's use `PairType::Custom("Anchor-XYK")`

Reply:
done